server/src/main/java/org/opensearch/rest/NamedRoute.java

```diff
@@ -41,15 +48,41 @@ public NamedRoute(RestRequest.Method method, String path, String name) {
         this.name = name;
     }
 
+    /**
+     * Allows registering a legacyName to match against transport action
+     * @param method - The REST method for this route
+     * @param path - the URL for this route
+     * @param name - the shortname for this route
+     * @param legacyActionName - name of the transport action this route will be matched against
+     */
+    public NamedRoute(RestRequest.Method method, String path, String name, String legacyActionName) {
+        this(method, path, name);
+        if (!TransportService.isValidActionName(legacyActionName)) {
+            logger.warn("invalid action name [" + legacyActionName + "] must start with one of: " + TransportService.VALID_ACTION_PREFIXES);
+        }
```

Comment:
We shouldn't allow these, just reject immediately for now.  We can make this less restrictive in time

Reply:
I agree with @peternied , but to provide additional context this is the what core currently does for transport action names.

See: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/transport/TransportService.java#L1130-L1136